logging/logging.go

```diff
@@ -65,23 +65,41 @@ func NewLogger(name string) Logger {
 		name:       name,
 		level:      NewAtomicLevelAt(INFO),
 		appenders:  []Appender{NewStdoutAppender()},
+		registry:   newRegistry(),
 		testHelper: func() {},
 	}
 
-	RegisterLogger(name, logger)
+	logger.registry.registerLogger(name, logger)
 	return logger
 }
 
+// NewLoggerWithRegistry is the same as NewLogger but also returns the
+// associated Registry.
+func NewLoggerWithRegistry(name string) (Logger, *Registry) {
```

Comment:
This is the new method we discussed offline. Used in entrypoint code to create the highest level `rdk` logger and update its associated registry whenever a new config comes in.